forgerock-openbanking-uk-aspsp-as/forgerock-openbanking-uk-aspsp-as-gateway/forgerock-openbanking-uk-aspsp-as-gateway-server/src/main/java/com/forgerock/openbanking/aspsp/as/service/TppRegistrationService.java

```diff
@@ -251,6 +258,27 @@ public void verifyTPPRegistrationRequestAgainstSSA(RegistrationRequest registrat
 
     public Tpp registerTpp(ApiClientIdentity clientIdentity, RegistrationRequest oidcRegistrationRequest)
             throws DynamicClientRegistrationException {
+        
+        Optional<DirectorySoftwareStatement> jti = tppStoreService.findByOrganisationId(clientIdentity.getUsername())
+            .stream()
+            .map(Tpp::getSsa)
+            .map(ssa -> {
+                try {
+                    return objectMapper.readValue(ssa, DirectorySoftwareStatement.class);
+                } catch (IOException e) {
+                    log.error("registerTpp() SSA cannot be converted to a DirectorySoftwareStatement", e);
+                }
+            })
+            .filter(dss -> dss.getJti().equals(oidcRegistrationRequest.getJti()))
```

Comment:
Not sure if this check is needed. 
- Check Dynamic Client Registration spec to see if it has any relevant info
- Check with customers to see what their prod system do